packages/core-flows/src/fulfillment/steps/cancel-fulfillments.ts

```diff
@@ -0,0 +1,18 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IFulfillmentModuleService } from "@medusajs/types"
+import { promiseAll } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+export const cancelFulfillmentsStepId = "cancel-fulfillments"
+export const cancelFulfillmentsStep = createStep(
+  cancelFulfillmentsStepId,
+  async (ids: string[], { container }) => {
+    const service = container.resolve<IFulfillmentModuleService>(
+      ModuleRegistrationName.FULFILLMENT
+    )
+
+    await promiseAll(ids.map((id) => service.cancelFulfillment(id)))
```

Comment:
@olivermrbl this looks like a non reversible step. Should we remove bulk capabilities here if that is the case?

Reply:
I think that makes sense, nice catch. We should treat all steps that interface with 3rd party providers this way – as we did with payments already. So bulk operations would have to be handled "manually" by the consumer

Reply:
nice, changed em!